lib/oban/query.ex

```diff
@@ -219,6 +219,32 @@ defmodule Oban.Query do
     :ok
   end
 
+  @spec force_retry(Config.t(), Job.t()) :: :ok
+  def force_retry(%Config{} = conf, %Job{} = job) do
+    %Config{prefix: prefix, repo: repo, verbose: verbose} = conf
+    %Job{attempt: attempt, id: id, max_attempts: max_attempts} = job
+
+    updates =
+      if attempt >= max_attempts do
+        [set: [state: "available", max_attempts: attempt + 1]]
```

Comment:
You can use `GREATEST` in a fragment to handle this update, rather than relying on data from the job.

Also of note, the state is changed to `available`, but the `completed_at` and `discarded_at` timestamps aren't cleared. That could allow this to run successfully yet still have a `discarded_at` timestamp.